### ISSUE TITLE
Add LintDiagnostic::severity

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno_lint#1274]
Other popular linters like ESLint allow to customize the severity of diagnostics - either "warning" or "error". Currently `deno_lint` treats all diagnostics all "error".

To allow for severity customization there are two major parts that need to happen:
1. Add `severity` field to `LintDiagnostic` struct so it can be properly formatted during printing
2. Add configuration options that would allow to customize the severity of each lint rule.

To handle point 1 following things need to happen:
1. Add `LintDiagnosticSeverity` enum like so:
```rust
#[derive(Debug, Default, Clone, Copy)]
pub enum LintDiagnosticSeverity {
  #[default]
  Error,
  Warning,
}
```
2. Update `LintDiagnostic` struct and add `severity: LintDiagnosticSeverity` field to it
3. Update necessary APIs to accept `severity: LintDiagnosticSeverity` argument in following order:
- `Context::create_diagnostic`
- `Context::add_diagnostic_with_fixes`
- `Context::add_diagnostic_with_hint`
- `Context::add_diagnostic`
4. Update `LintDiagnostic::level()` to convert `LintDiagnosticSeverity` into `deno_ast::DiagnosticLevel`.

These changes should be enough to be able to apply different formatting based on severity and can be first applied in `examples/dlint/diagnostics.rs` in `print_compact` and `print_pretty`.

Handling point 2 is more involved and requires further discussion as the current lint configuration is rather bare-bones - it allows to specify a list of lints to run, but not customize them one-by-one.

If someone picks up this issue I can provide more details on how to handle that, but mind that it require changes across several repositories - `deno_lint`, `deno_config` and `deno` itself.

**Comment by eryue0220** [comment]
Could I have a try for this one if no one take it?

**Comment by bartlomieju** [comment]
@eryue0220 sorry for a slow response. This is still up for grabs if you want to tackle it.
